src/journal-remote/journal-upload.c

```diff
@@ -520,9 +526,53 @@ static int perform_upload(Uploader *u) {
                 return log_error_errno(SYNTHETIC_ERRNO(EIO),
                                        "Upload to %s finished with unexpected code %ld: %s",
                                        u->url, status, strna(u->answer));
-        else
+        else {
+                if (u->compression->algorithm == COMPRESSION_NONE) {
+                        Compression algorithm;
+                        bool found = false;
+                        hcode = curl_easy_header(u->easy, "Accept-Encoding", 0, CURLH_HEADER, -1, &encoding);
+                        if (hcode == CURLHE_OK && encoding && encoding->value) {
+                                while(encoding->value != NULL && !found) {
+                                        _cleanup_free_ char *encVal = NULL;
+                                        _cleanup_free_ char *a = NULL;
+
+                                        if (extract_first_word((const char **)&encoding->value, &encVal, ",", 0) < 0)
```

Comment:
no, the first param takes a pointer we advance to the next word, but you shouldn't advance `value` here, no?